### PR TITLE
Add RD4AD model

### PR DESCRIPTION
**Pull request by LukaszMarcjan**
## Summary

Adds the **RD4AD** model. It builds on the shared vision base introduced in the FastFlow PR, implemented as a thin `LightningVisionModel` subclass.

## What's included

- **RD4AD model**: reverse-distillation architecture (frozen ResNet teacher, one-class bottleneck embedding, reverse-ResNet student decoder), cosine-similarity distillation loss, and an additive multi-scale anomaly map.
- **Config** with defaults (WideResNet-50-2 backbone, 256×256 input, Adam `lr=5e-3` / betas `(0.5, 0.999)`, 200 epochs).
- **Concept-incremental example** and tests.

### CODE REVIEW

File: src/pyclad/vision/models/rd4ad/config.py [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+    learning_rate: float = Field(default=5e-3, gt=0.0)
+
+    backbone_name: Literal["resnet18", "resnet34", "resnet50", "wide_resnet50_2"] = "wide_resnet50_2"
+    pretrained_encoder: bool = True
+    freeze_encoder: bool = True
+
+    adam_beta1: float = Field(default=0.5, ge=0.0, lt=1.0)
+    adam_beta2: float = Field(default=0.999, ge=0.0, lt=1.0)
+
+    score_smoothing_sigma: float = Field(default=0.0, ge=0.0)
```
Comment (Nyderx) [inline src/pyclad/vision/models/rd4ad/config.py:23]: Should we keep the default here as 0? It disables smoothing, right? Maybe we should use the hyperparameter value from original paper/code?

**Review by Nyderx — CHANGES_REQUESTED** [review]

File: src/pyclad/vision/models/rd4ad/config.py (hunk 1 above)
Comment (LukaszMarcjan) [reply src/pyclad/vision/models/rd4ad/config.py]: 0 disabled smoothing entirely. The original paper code applies sigma=4 on the evaluation path that produces the reported metrics, and every other default here already matched the paper, so this was the only outlier. Changed the default to 4.0.

**Review by Nyderx — DISMISSED** [review]

**Review by Nyderx — DISMISSED** [review]

**Review by Nyderx — APPROVED** [review]

**Review by Nyderx — APPROVED** [review]
